### PR TITLE
refactor norm-type/statistics mlm:input properties

### LINKED ISSUE #26: Add explicit dependency checks for normalization details
**[ @fmigneault](https://github.com/fmigneault)** cloned issue [crim-ca/mlm-extension#10](https://github.com/crim-ca/mlm-extension/issues/10) on 2024-05-01: 

> ## :rocket: Feature Request
> 
> 
> 
> Within a `mlm:input` definition, when a `norm_type` is specified, it is expected that the corresponding `statistics` details are provided for the relevant normalization technique.
> 
> 
> 
> For example, a `z-score` needs the corresponding `mean` and `stddev` (minimally), while `min-max` would require the `minimum` and `maximum` statistics. 
> 
> 
> 
> For the JSON schema, the `dependencies` property could be used to check that when `norm_type` is equal to some constant, the subset of corresponding `statistics` are defined + `minItems: 1` or similar. In the `stac_model` definition, a `@model_validator(mode="after")` on the `ModelInput` could be applied.
> 
> 
> 
> ## :sound: Motivation
> 
> 
> 
> Ensure that the intended definition is properly specified.
> 
> If the user forgot to apply some parameters unintentionally, this check would ensure the error is caught early, increasing model reusability. Not all cases can necessarily be covered, but most common normalization techniques could be handled. 
> 
> 
> 
> ## :satellite: Alternatives
> 
> 
> 
> There are no explicit validation currently.
> 
> 
> 
> ## :paperclip: Additional context
> 
> 
> 
> n/a
> 
>

### PR DESCRIPTION
**Pull request by fmigneault**
## Description

- Adds the `scaling` field to replace the combination of `norm_type`, `norm_clip`, `norm_by_channel` and `statistics` field combinations under a common definition.
- Adds better validation of expected sub-fields by each `scaling` type. 

## Related Issue

- fixes #26 

## Type of Change

- [ ] :books: Examples, docs, tutorials or dependencies update;
- [ ] :wrench: Bug fix (non-breaking change which fixes an issue);
- [ ] :clinking_glasses: Improvement (non-breaking change which improves an existing feature);
- [ ] :rocket: New feature (non-breaking change which adds functionality);
- [x] :boom: Breaking change (fix or feature that would cause existing functionality to change);
- [ ] :closed_lock_with_key: Security fix.

## Checklist

- [x] I've read the [`CONTRIBUTING.md`](../CONTRIBUTING.md) guide;
- [ ] I've updated the code style using `make check`;
- [x] I've written tests for all new methods and classes that I created;
- [ ] I've written the docstring in `Google` format for all the methods and classes that I used.

### CODE REVIEW

File: stac_model/input.py [hunk 1]
```diff
@@ -18,27 +18,58 @@ def validate_dimensions(self) -> Self:
...
+ScalingObject: TypeAlias = Optional[
+    Union[
+        ScalingMinMax,
+        ScalingZScore,
+        ScalingClip,
+        ScalingClipMin,
+        ScalingClipMax,
+        ScalingOffset,
+        ScalingScale,
+        ScalingProcessingExpression,
```
Comment (rbavery) [inline stac_model/input.py:72]: Is it confusing for users to be able to provide processing expressions in two different locations within the ModelInputObject? I thought that processing expressions would still only be supplied via the existing pre_processing_function field?

**Review by rbavery — CHANGES_REQUESTED** [review]

File: stac_model/input.py (hunk 1 above)
Comment (fmigneault) [reply stac_model/input.py:72]: I'm guessing using processing expressions is for advanced use cases anyway, so I would expect only more advanced users to define them anyway. It at least offers some flexibility for overall-resize/scaling vs per-band-resize/scaling expressions. Before, per-band expression would have needed to be embedded within the overall expression, which made it itself way more convoluted.

File: json-schema/schema.json [hunk 2]
```diff
@@ -490,46 +490,40 @@
...
+          "$ref": "#/$defs/ModelBands"
+        },
+        "input": {
+          "$ref": "#/$defs/InputStructure"
+        },
+        "description": {
+          "type": "string",
+          "minLength": 1
+        },
+        "scaling": {
```
Comment (rbavery) [inline json-schema/schema.json:519]: I'm having some lingering concern that "scaling" will be confused with spatial scale. what do you think of renaming this to "value_scaling"?

Comment (fmigneault) [reply json-schema/schema.json:519]: yes, we can do that

Comment (rbavery) [reply json-schema/schema.json:519]: ok awesome. good to merge after rename.

Comment (fmigneault) [reply json-schema/schema.json]: Changed. Please review again just to make sure I didn't mess up any search/replace.

**Review by rbavery — APPROVED** [review]
Looks good! looking forward to clearer semantics for applying statistics (no value_scaling) to inputs.

**Pull request merged**